Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -291,3 +385,224 @@ private extension CardReaderConfigError {
         }
     }
 }
+
+// MARK: Networking Methods
+private extension CardPresentPaymentStore {
+    /// Switch the store to use WCPay as the backend.
+    /// Does nothing if the store is already using WCPay.
+    /// WCPay is the initial default for the store.
+    ///
+    func useWCPayAsBackend() {
+        guard usingBackend != .wcpay else {
+            return
+        }
+
+        reset() // Concern: asynchronicity
+
+        usingBackend = .wcpay
+        print("==== Switched CardPresentPaymentStore to use WCPay as the backend")
+    }
+
+    /// Switch the store to use Stripe as the backend.
+    /// Does nothing if the store is already using Stripe.
+    ///
+    func useStripeAsBackend() {
+        guard usingBackend != .stripe else {
+            return
+        }
+
+        reset() // Concern: asynchronicity
+
+        usingBackend = .stripe
+        print("==== Switched CardPresentPaymentStore to use Stripe as the backend")
+    }
+
+    /// Loads the account corresponding to the currently selected backend. Deletes the other (if it exists).
+    ///
+    func loadAccounts(siteID: Int64, onCompletion: @escaping (Result<Void, Error>) -> Void) {
+        switch usingBackend {
+        case .wcpay:
+            print("==== CPPS loadAccounts WCPay")
+            loadWCPayAccount(siteID: siteID, onCompletion: onCompletion)
+        case .stripe:
+            print("==== CPPS loadAccounts Stripe")
+            loadStripeAccount(siteID: siteID, onCompletion: onCompletion)
+        }
+    }
+
+    func loadWCPayAccount(siteID: Int64, onCompletion: @escaping (Result<Void, Error>) -> Void) {
+        /// Delete any Stripe account present. There can be only one.
+        self.deleteStaleAccount(siteID: siteID, gatewayID: StripeAccount.gatewayID)
+
+        /// Fetch the WCPay account
+        remote.loadAccount(for: siteID) { [weak self] result in
+            guard let self = self else {
+                return
+            }
+
+            switch result {
+            case .success(let wcpayAccount):
+                let account = wcpayAccount.toPaymentGatewayAccount(siteID: siteID)
+                print("==== CPPS successfully fetched WCPay account")
+                self.upsertStoredAccountInBackground(readonlyAccount: account)
+            case .failure:
+                print("==== CPPS was unable to fetch WCPay account")
+                DDLogDebug("Error fetching WCPay account - it is possible the extension is not installed or inactive")
+                self.deleteStaleAccount(siteID: siteID, gatewayID: WCPayAccount.gatewayID)
+            }
+
+            onCompletion(.success(())) // TODO - consider bringing back Error, although callers don't currently use it
```

Comment:
Please don't make this a TODO, make it a ticket. Even if clients don't use it, it is good to surface as many errors as possible

Reply:
I just went ahead and brought back failure - https://github.com/woocommerce/woocommerce-ios/pull/5868/commits/5c19233add147d6e8ac7b59c98533e66999fe17e